### DISCUSSION TITLE
Update minimum PHP requirements

### DISCUSSION THREAD

**Discussion by avoidthehack** [JulianPrieber/littlelink-custom#648] [Questions]
Hi!

PHP 8.0 is completely deprecated as of Nov 2023. It is no longer receiving any security updates: https://endoflife.date/php

Recommend updating dependency/system checks for at bare minimum PHP 8.1 - it still has security updates for at least another year before it is completely deprecated as well. Ideally, it should be 8.2 as it has active support for at least another year, security support for 2 years.

**Comment by JulianPrieber** [comment]
LinkStack is fully compatible up to PHP 8.3 and our dependencies reflect that. We follow our own timeline, we may require at least PHP 8.1 in Q1 2024 and 8.2 from 2025 onwards.

I'd prefer if we would still support PHP 7.4, but we do make sure we are getting active security updates.

The docker version is running PHP 8.2. All our hosted servers are running at least PHP 8.2.

**Reply by avoidthehack** [reply]
I meant more for the folks who are hosting their own, whether that is on a VPS or otherwise and not necessarily using Docker (I'm not, personally). I have it also running on PHP 8.2 and it runs flawlessly, so no issues there.

Just trying to account for the lowest common denominator, is all. Folk using managed hosting services naturally don't have to worry (well, hope they don't) as their providers will likely keep up with PHP versions on their servers. 

Figured it would be an easy (lol) change to bump up minimum to 8.1 to just as an overall improvement. If you have plans to do so in Q1 2024, that works too.

**Reply by JulianPrieber** [reply]
Right, I get that. The first sentence in our security policy is "The security of this application is left to the user." and that's mostly our approach, unless it's something a reasonably thinking admin wouldn't know.

I mostly like to maintain compatibility going as far back as possible. If someone wants to run PHP 5 they should be allowed to.

We mostly only do switch due to feature or security support to keep our codebase up to date. Last year we did the same with PHP 7.4.
I just hate to see people not being able to use our app because their system is slightly outdated. 

2022, PHP 7.4 still had a market share of what? Almost 40%? When PHP 7.4 was EOL, a large portion shifted to newer versions. People don't like to update.

Just because security support dropped, there's not immediately gonna be a Log4J level threat.

I want to maintain support for legacy systems for as long as possible, only switching when required. We'll definitely switch, it might just take a bit.

**Reply by avoidthehack** [reply]
I missed that in your security policy. 

If backwards compatibility (within reason) is your focus, that's fine. Outdated minimums still kind of make me recoil, but that could be because I work in security. Naturally, the security burden rests on the admin but... ah, I digress.

In any case, perhaps it could just be an item to include in the docs for those interested. I still just wanted to flag it in any case.

**Reply by JulianPrieber** [reply]
Riight, I just looked at your profile. I saw your post on Mastodon like a week ago.
If you don't mind me asking, what field in security are you working in?
Any chance you'd be open to helping us out in the future? We're looking for someone with expertise in security to join our volunteer team as a consultant.

**Reply by avoidthehack** [reply]
Hmm. I work in threat intelligence. I'm open to the idea of helping out, but not quite sure how much help I can really be. Web AppSec kind of came with the territory of running my own web server, but it's not what I do in my day-to-day.

**Reply by JulianPrieber** [reply]
We mostly need help with Pentests. Otherwise, general threat detection and assessment.
We can't really offer any bug bounties, so we are dependent on the help from the open source community.
We need someone who knows how to perform more specialized attacks/use specialized tools.

**Reply by avoidthehack** [reply]
I see. That's a little outside what I do in a day-today (though, it is something I'm always learning). Ah, maybe one day. I think I'm going to stay the threat intel route though, haha.
